service/gcs/core/gcs/networking.go

```diff
@@ -1,146 +1,63 @@
 package gcs
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
 	"os"
 	"path/filepath"
 	"strings"
-	"syscall"
 
-	"github.com/Microsoft/opengcs/service/gcs/oslayer"
 	"github.com/Microsoft/opengcs/service/gcs/prot"
 	"github.com/Microsoft/opengcs/service/gcs/runtime"
+	"github.com/Sirupsen/logrus"
 	"github.com/pkg/errors"
 )
 
-// instanceIDToName converts from the given instance ID (a GUID generated on
-// the Windows host) to its corresponding interface name (e.g. "eth0").
-func (c *gcsCore) instanceIDToName(id string) (string, error) {
-	deviceDirs, err := c.OS.ReadDir(filepath.Join("/sys", "bus", "vmbus", "devices", id, "net"))
-	if err != nil {
-		return "", errors.Wrapf(err, "failed to read vmbus network device from /sys filesystem for adapter %s", id)
-	}
-	if len(deviceDirs) == 0 {
-		return "", errors.Errorf("no interface name found for adapter %s", id)
-	}
-	if len(deviceDirs) > 1 {
-		return "", errors.Errorf("multiple interface names found for adapter %s", id)
-	}
-	return deviceDirs[0].Name(), nil
-}
-
-// getLinkForAdapter returns the oslayer.Link corresponding to the given
-// network adapter.
-func (c *gcsCore) getLinkForAdapter(adapter prot.NetworkAdapter) (oslayer.Link, error) {
+// configureAdapterInNamespace moves a given adapter into a network
+// namespace and configures it there.
+func (c *gcsCore) configureAdapterInNamespace(container runtime.Container, adapter prot.NetworkAdapter) error {
 	id := adapter.AdapterInstanceID
 	interfaceName, err := c.instanceIDToName(id)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	link, err := c.OS.GetLinkByName(interfaceName)
+	nspid := container.Pid()
+	cfg, err := json.Marshal(adapter)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get network interface for adapter %s", id)
+		return errors.Wrapf(err, "failed to marshal adapter struct to JSON for adapter %s", id)
 	}
-	return link, nil
-}
 
-// configureNetworkAdapter configures the given network adapter to be used by
-// the container.
-func (c *gcsCore) configureNetworkAdapter(adapter prot.NetworkAdapter) error {
-	link, err := c.getLinkForAdapter(adapter)
+	out, err := c.OS.Command("netnscfg",
```

Comment:
it looks like this change requiring a new tool netnscfg, any particular version is required?